apps/passport-client/src/dispatch.ts

```diff
@@ -576,17 +576,61 @@ async function makeUploadId(
  *   to e2ee, then uploads then to e2ee.
  */
 async function sync(state: AppState, update: ZuUpdate) {
+  // This re-entrancy protection ensures only one event-sequence is
+  // inside of doSync() at any time.  When doSync() completes, it will
+  // run again if any state changes were made, or if any updates were skipped.
+  if (!syncInProgress) {
+    try {
+      syncInProgress = true;
+      const stateChanges = await doSync(state, update);
+      if (stateChanges) {
+        update(stateChanges);
+      } else if (skippedSyncUpdates > 0) {
+        console.log("[SYNC] running an extra sync in case of missed updates");
+        update({});
```

Comment:
i am assuming you're using this line of code to trigger another sync? this is probably not super clear to the uninitiated